src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,156 @@
+export class AzureFunctionProjectService {
+
+	constructor(private azureFunctionsService: AzureFunctionsService) {
+	}
+
+	public async createAzureFunction(connectionString: string, schema: string, table: string): Promise<void> {
+		const azureFunctionApi = await this.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			return;
+		}
+		if (!await this.isAzureFunctionProjectOpen()) {
+			vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened);
+			return;
+		}
+
+		// because of an AF extension API issue, we have to get the newly created file by adding
+		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
+		const newFilePromise = this.getNewFunctionFile();
+
+		// get function name from user
+		const functionName = await vscode.window.showInputBox({
+			title: 'Function Name',
+			value: 'HttpTrigger1'
+		});
+
+		await azureFunctionApi.createFunction({
+			language: 'C#',
+			templateId: 'HttpTrigger',
+			functionName: functionName
+		});
+
+		await this.addNugetReferenceToProjectFile();
+		await this.addConnectionStringToConfig(connectionString);
+		const functionFile = await newFilePromise;
+
+
+		await this.azureFunctionsService.addSqlBinding(
+			mssql.BindingType.input,
+			functionFile,
+			functionName,
+			schema + '.' + table,
+			sqlConnectionString
+		);
+
+		this.refactorAzureFunction(functionFile);
+	}
+
+	private async getAzureFunctionsExtensionApi(): Promise<af.AzureFunctionsExtensionApi> {
+		const afExtension = vscode.extensions.getExtension(azureFunctionsExtensionName);
+		if (afExtension) {
+			let azureFunctionApi;
+			if (!afExtension.isActive) {
+				azureFunctionApi = await afExtension.activate();
+			} else {
+				azureFunctionApi = afExtension.exports;
+			}
+			return azureFunctionApi.getApi('*') as af.AzureFunctionsExtensionApi;
+		} else {
+			vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsExtensionNotInstalled);
+			return undefined;
+		}
+	}
+
+	private refactorAzureFunction(filePath: string): void {
+		let defaultBindedFunctionText = fs.readFileSync(filePath, 'utf-8');
+		// Add missing import for Enumerable
+		let newValue = genericCollectionImport + os.EOL + defaultBindedFunctionText;
+		// Replace default binding text
+		let newValueLines = newValue.split(os.EOL);
+		const defaultLineSet = new Set(defaultSqlBindingTextLines);
+		let replacedValueLines = [];
+		for (let defaultLine of newValueLines) {
+			// Skipped lines
+			if (defaultLineSet.has(defaultLine.trimStart())) {
+				continue;
+			} else if (defaultLine.trimStart() === defaultBindingResult) { // Result change
+				replacedValueLines.push(defaultLine.replace(defaultBindingResult, sqlBindingResult));
+			} else {
+				// Normal lines to be included
+				replacedValueLines.push(defaultLine);
+			}
+		}
+		newValue = replacedValueLines.join(os.EOL);
+		fs.writeFileSync(filePath, newValue, 'utf-8');
+	}
+
+	private async isAzureFunctionProjectOpen(): Promise<boolean> {
```

Comment:
All these helper functions can just be at the module level, no reason to have them in the class.